### PR TITLE
tried to refactor the auth functions

### PR DESCRIPTION
**Pull request by Kepmon**
I implemented this function in the way we were talking about but something is off - the functions (signing in and up) seem to work but some error occurs. But I can see nothing in the console, I know there is some error only because of the color and content of the toast.

By the way, don't pay attention to the content of the error toast in the sign-in page - I just assumed the only possibility some error would occur in here is when a user provides incorrect data.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -19,6 +17,7 @@ module.exports = {
   rules: {
     semi: ['error', 'never'],
     'no-trailing-spaces': ['error', { skipBlankLines: true }],
-    'comma-dangle': ['error', 'never']
+    'comma-dangle': ['error', 'never'],
+    'linebreak-style': ['error', 'windows']
```
Comment (aleksanderwalczuk) [inline .eslintrc.js:21]: Please set unix as it's more consistent with other OSes and supported by windows anyway afaik

Comment (aleksanderwalczuk) [reply .eslintrc.js:21]:
```suggestion
    'linebreak-style': ['error', 'unix']
```

File: src/stores/user.js [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
 export const useUserStore = defineStore('user', {
   state: () => ({
     isSignedIn: false
   }),
   actions: {
-    async signUp(email, password) {
+    async authenticateUser(email, password, method) {
       try {
-        const { error } = await supabase.auth.signUp({
+        const { error } = await method({
```
Comment (aleksanderwalczuk) [inline src/stores/user.js]: This won't work. When supabase methods are passed the lose context

File: .eslintrc.js (hunk 1 above)
Comment (Kepmon) [reply .eslintrc.js:21]: Unfortunately, your suggestion doesn't work on windows:
eslint

It wouldn't be such a big deal if I had to change it once but this comes back everytime I switch branches which is so annoying. I changed all VSC setting related to the default line break but they don't apply, don't know why.

File: src/stores/user.js (hunk 2 above)
Comment (Kepmon) [reply src/stores/user.js]: I understand, I change those functions back, then.

**Review by aleksanderwalczuk — APPROVED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (aleksanderwalczuk) [reply .eslintrc.js:21]: Did you set in the bottom right corner the end of line style to CRLF?

**Pull request merged**